### PR TITLE
pvs-studio: init at 7.33.85330.89

### PR DESCRIPTION
**Pull request by paveloom**
## Description of changes

https://pvs-studio.com/en/pvs-studio

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbsds** [comment]
```
pvs-studio-7.27.75620.346-x86_64.tgz> trying https://cdn.pvs-studio.com/pvs-studio-7.27.75620.346-x86_64.tgz
pvs-studio-7.27.75620.346-x86_64.tgz>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
pvs-studio-7.27.75620.346-x86_64.tgz>                                  Dload  Upload   Total   Spent    Left  Speed
pvs-studio-7.27.75620.346-x86_64.tgz>   0   146    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
pvs-studio-7.27.75620.346-x86_64.tgz> curl: (56) The requested URL returned error: 404
pvs-studio-7.27.75620.346-x86_64.tgz> error: cannot download pvs-studio-7.27.75620.346-x86_64.tgz from any mirror
```

File: pkgs/by-name/pv/pvs-studio/package.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  src =
+    let
+      system = stdenv.hostPlatform.system;
+      selectSystem = attrs:
+        attrs.${system} or (throw "Unsupported system: ${system}");
+    in
+    fetchurl {
+      url = selectSystem {
+        x86_64-darwin = "https://cdn.pvs-studio.com/pvs-studio-${version}-macos.tgz";
+        x86_64-linux = "https://cdn.pvs-studio.com/pvs-studio-${version}-x86_64.tgz";
```
Comment (FliegendeWurst) [inline pkgs/by-name/pv/pvs-studio/package.nix:23]: No longer available for that version.

File: pkgs/by-name/pv/pvs-studio/package.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
```
Comment (FliegendeWurst) [inline pkgs/by-name/pv/pvs-studio/package.nix:2]: Please format with new formatter

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/pv/pvs-studio/package.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/by-name/pv/pvs-studio/package.nix:23]: urls going 404 is a problem, especially for unfree. Consider using archive.org

Comment (paveloom) [reply pkgs/by-name/pv/pvs-studio/package.nix:23]: Archiving both via [archive.org](https://web.archive.org) and [archive.is](https://archive.is) didn't work.

Comment (paveloom) [reply pkgs/by-name/pv/pvs-studio/package.nix]: Okay, I just noticed the archives actually appeared on the Wayback Machine after a while.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**